segmentationRDS/image.py

```diff
@@ -53,11 +53,67 @@ def srgb_gamma_inv(x):
     else:
         return 1.055 * pow(min(1.0, x), 1.0/2.4) - 0.055
 
-def loadImage(imagePath: str, applyPAR: bool = False, incolorspace: str = 'acescg'):
+def find_metadata(oiio_spec, name: str, default):
+    values = []
+    oiio_extra_attribs = oiio_spec.extra_attribs
+    for i in range(len(oiio_extra_attribs)):
+        pos = oiio_extra_attribs[i].name.find(name)
+        if pos == 0:
+            values.insert(0, oiio_spec.getattribute(oiio_extra_attribs[i].name))
+        elif pos != -1:
+            values.append(oiio_spec.getattribute(oiio_extra_attribs[i].name))
+    if len(values) == 0:
+        values.append(default)
+    return values
+
+def apply_orientation(oiio_image, orientation, reverse: bool = False):
```

Comment:
Couldn't we use oiio's native function?
https://openimageio.readthedocs.io/en/stable/imagebufalgo.html#_CPPv4N4OIIO12ImageBufAlgo8reorientERK8ImageBufi

Reply:
We cannot use the oiio reorient function when writing the result images (mask or tags) because the metadata "Orientation" is set to 1 after updating the image but we want to keep it equal to the value in the source image.

Reply:
```
def apply_orientation(oiio_image, orientation, reverse: bool = False):
    img.set_orientation(orientation)
    img = ImageBufAlgo.reorient(img)
    img.specmod().attribute("AliceVision:origOrientation", orientation)
```

As it's done for RAW images:
```
// libraw reoriented the image for us, so squash any orientation
// metadata we may have found in the Exif. Preserve the original as
// "raw:Orientation".
```

For maintainance, I think that it would be important to avoid the duplication of the logic on the orientation flag interpretation.

Reply:
Except that when we write the result we want the opposite of what is done by the oiio reorient function. The reorient function use the embedded metadata to rotate the image and set the metadata to 1. In our case the metadata value is 1 and we want a reorientation to match the one of the source image, with an embedded metadata that is not 1. The oiio function does not allow that.
If we don't care about the matching of the Orientation metadata value of the mask image with the one of the source image, we can use the oiio function. In this case, the mask image Orientation metadata will be always set to 1.